## Compare FTD labels with GINCO

Based on the FTD predictions that we applied to the GINCO, I'll analyse how the FTD labels are connected with GINCO labels.

In [1]:
import pandas as pd
import numpy as np

In [3]:
# Import the GINCO dataset
ginco = pd.read_csv("final_data/GINCO-MT-GINCO-keeptext-split-file-with-all-information.csv", sep="\t", index_col=0)
ginco.head(3)

id                                                url  crawled   hard  \
0  3949  http://www.pomurje.si/aktualno/sport/zimska-li...     2014  False   
1  3726  http://www.ss-sezana.si/sss/index.php?option=c...     2014  False   
2  5621  http://www.kamnik-starejsi.si/novice/144-sodel...     2014  False   

           primary_level_1          primary_level_2          primary_level_3  \
0           News/Reporting           News/Reporting           News/Reporting   
1  Information/Explanation  Information/Explanation  Information/Explanation   
2    Promotion of Services    Promotion of Services                Promotion   

       secondary_level_1      secondary_level_2      secondary_level_3  ...  \
0                    NaN                    NaN                    NaN  ...   
1                    NaN                    NaN                    NaN  ...   
2  Opinion/Argumentation  Opinion/Argumentation  Opinion/Argumentation  ...   

          tertiary_level_3  split                  domain  \
0                      NaN   test          www.pomurje.si   
1                      NaN  train        www.ss-sezana.si   
2  Information/Explanation  train  www.kamnik-starejsi.si   

                   GINCORE                                       Slovene_text  \
0                     News  Šport <p/> Zimska liga malega nogometa sobota,...   
1  Information/Explanation  JEDILNIK <p/> Iskalnik <p/> Poglavitni cilj pr...   
2                Promotion  Projekt INNOVAge in zavod Oreli <p/> Zavod Ore...   

                                             MT_text text_length  \
0  Sport <p/> Winter Little League Football Satur...          93   
1  JEDILNIK <p/> Search <p/> The main objective o...          76   
2  Project INNOVAge and the Oreli Institute <p/> ...         232   

      FTD_pred_on_SL     FTD_pred_on_MT split-without-rare-categories  
0          A8 (news)          A8 (news)                          test  
1  A16 (information)  A16 (information)                          test  
2    A12 (promotion)    A12 (promotion)                         train  

[3 rows x 22 columns]

In [6]:
# See label distribution in GINCO
ginco.primary_level_1.value_counts()

Information/Explanation       130
News/Reporting                115
Promotion of a Product        115
Opinion/Argumentation         114
List of Summaries/Excerpts    106
Opinionated News               89
Forum                          52
Instruction                    38
Other                          34
Invitation                     32
Promotion of Services          32
Promotion                      30
Legal/Regulation               17
Announcement                   17
Review                         17
Correspondence                 16
Call                           11
Research Article                9
Interview                       8
Recipe                          6
Prose                           6
Lyrical                         4
FAQ                             3
Script/Drama                    1
Name: primary_level_1, dtype: int64

In [9]:
# Analyse how GINCO primary labels are connected with FTD labels based on the prediction
pd.crosstab(ginco["primary_level_1"], ginco["FTD_pred_on_MT"], normalize="index")


FTD_pred_on_MT              A1 (argumentative)  A11 (personal)  \
primary_level_1                                                  
Announcement                          0.000000        0.000000   
Call                                  0.272727        0.000000   
Correspondence                        0.312500        0.187500   
FAQ                                   0.000000        0.000000   
Forum                                 0.384615        0.403846   
Information/Explanation               0.046154        0.007692   
Instruction                           0.000000        0.026316   
Interview                             0.250000        0.375000   
Invitation                            0.000000        0.000000   
Legal/Regulation                      0.000000        0.000000   
List of Summaries/Excerpts            0.141509        0.047170   
Lyrical                               0.000000        0.750000   
News/Reporting                        0.069565        0.000000   
Opinion/Argumentation                 0.456140        0.184211   
Opinionated News                      0.202247        0.000000   
Other                                 0.117647        0.088235   
Promotion                             0.000000        0.000000   
Promotion of Services                 0.031250        0.000000   
Promotion of a Product                0.000000        0.000000   
Prose                                 0.000000        0.166667   
Recipe                                0.000000        0.166667   
Research Article                      0.111111        0.000000   
Review                                0.058824        0.117647   
Script/Drama                          0.000000        0.000000   

FTD_pred_on_MT              A12 (promotion)  A14 (academic)  \
primary_level_1                                               
Announcement                       0.411765        0.000000   
Call                               0.272727        0.000000   
Correspondence                     0.125000        0.000000   
FAQ                                0.333333        0.000000   
Forum                              0.057692        0.000000   
Information/Explanation            0.253846        0.000000   
Instruction                        0.131579        0.000000   
Interview                          0.250000        0.000000   
Invitation                         0.812500        0.000000   
Legal/Regulation                   0.000000        0.000000   
List of Summaries/Excerpts         0.264151        0.009434   
Lyrical                            0.000000        0.000000   
News/Reporting                     0.034783        0.008696   
Opinion/Argumentation              0.070175        0.000000   
Opinionated News                   0.044944        0.000000   
Other                              0.294118        0.000000   
Promotion                          0.800000        0.000000   
Promotion of Services              0.906250        0.000000   
Promotion of a Product             0.826087        0.000000   
Prose                              0.000000        0.000000   
Recipe                             0.000000        0.000000   
Research Article                   0.000000        0.777778   
Review                             0.176471        0.000000   
Script/Drama                       0.000000        0.000000   

FTD_pred_on_MT              A16 (information)  A17 (review)  A4 (fiction)  \
primary_level_1                                                             
Announcement                         0.117647      0.000000      0.000000   
Call                                 0.000000      0.000000      0.000000   
Correspondence                       0.000000      0.062500      0.000000   
FAQ                                  0.000000      0.000000      0.000000   
Forum                                0.038462      0.019231      0.000000   
Information/Explanation              0.569231      0.015385      0.007692   
Instruction                 

In [ ]:
pd.crosstab(ginco["primary_level_1"], ginco["FTD_pred_on_MT"], normalize="index").to_dict("index")

What we can see based on the FTD predictions on the GINCO labels, is (first, the information for prediction on Slovene text is given, followed by the information what is different on predictions on the MT text):

1. Some labels are very well connected (similar):
    - Instruction (38): very well identified as 'A7 (instruction)': 0.79; on MT text even better identified: 'A7 (instruction)': 0.84
    - Invitation (32): very well identified as  'A12 (promotion)': 0.875; on MT slightly worse identified as 'A12 (promotion)': 0.81, 20% identified as News: 'A8 (news)': 0.1875,
    - Legal/Regulation (17): very well identified as 'A9 (legal)': 1.0, on MT the same
    - Promotion of Services (32): very well predicted as  'A12 (promotion)': 0.94; a bit worse predicted on MT: 'A12 (promotion)': 0.91
    - Promotion of a Product (115): very well predicted as 'A12 (promotion)': 0.85; a bit worse predicted on MT: 'A12 (promotion)': 0.83
    - Promotion (30): very well predicted as  'A12 (promotion)': 0.8; the same on MT
    - Prose (6): very well predicted as 'A4 (fiction)': 1.0; on MT a bit worse predicted: 'A4 (fiction)': 0.83, 'A11 (personal)': 0.17
    - Recipe (6): very well predicted as 'A7 (instruction)': 0.83; the same on MT
    - Research Article (9): well predicted as  'A14 (academic)': 0.67, otherwise as 'A16 (information)': 0.22; on MT even better prediction: 'A14 (academic)': 0.77, otherwise 'A16 (information)': 0.11
    - Review (17): predicted as 'A17 (review)': 0.65, otherwise 'A12 (promotion)': 0.29; on MT similar prediction of review, but less promotion: 'A12 (promotion)': 0.17, some 'A11 (personal)': 0.12 instead

2. Some seem to be less connected:
    - Information/Explanation (130 instances): relatively well identified as 'A16 (information)': 0.6, a third identified as  'A12 (promotion)': 0.32; on MT text similarly identified as 'A16 (information)': 0.57, 25% identified as 'A12 (promotion)'.
    - News/Reporting (115): half identified as 'A8 (news)': 0.55, otherwise also 'A1 (argumentative)': 0.10,  'A12 (promotion)': 0.13,   'A16 (information)': 0.10 and other categories; on MT text much better identification of News: 'A8 (news)': 0.79
    - Opinion/Argumentation (114): less than half identified as 'A1 (argumentative): 0.39, otherwise 'A11 (personal)': 0.14,  'A12 (promotion)': 0.12,  'A16 (information)': 0.17,  'A17 (review)': 0.13 and others; on MT text similar: mostly 'A1 (argumentative)': 0.46, 'A11 (personal)': 0.18
    - Opinionated News (89): half identified as 'A8 (news)': 0.528, otherwise mostly 'A1 (argumentative)': 0.13 or 'A12 (promotion)': 0.12 or  'A17 (review)': 0.17; on MT even more identified as News: 'A8 (news)': 0.63, 20% as  'A1 (argumentative)'


3. Some GINCO labels were predicted by a variety of FTD labels - are not well connected:
    - Announcement (17 instances): not closely connected to just one FTD label - instances were predicted as 'A12 (promotion)': 0.47 (of all Announcement instances), 'A16 (information)': 0.18, 'A7 (instruction)': 0.06,   'A8 (news)': 0.29; on MT text, the main label are 'A8 (news)': 0.41 and 'A12 (promotion)': 0.41
    - Call (11 instances): mostly connected to 'A12 (promotion)': 0.73; on MT text, it is divided between 'A1 (argumentative)', 'A12 (promotion)' and 'A9 (legal)' (0.27)
    - Correspondence (16 instances): not well predicted with FTDs - mostly split between 'A1 (argumentative)': 0.375, 'A7 (instruction)': 0.125, and 'A8 (news)': 0.0625 - on MT similar, but there is more 'A7 (instruction)': 0.25 and also 13% 'A12 (promotion)'
    - FAQ (3 instances): mostly predicted as 'A12 (promotion)': 0.67 - to note: there are very little FAQ instances in the corpus. On MT text, it is mostly predicted as 'A7 (instruction)': 0.66 and 33 % as 'A12 (promotion)'.
    - Forum (52 instances): not well connected with FTD categories - identified as 'A1 (argumentative)': 0.38,  'A11 (personal)': 0.19,'A17 (review)': 0.15, also some   'A12 (promotion)', 'A14 (academic)',  'A16 (information)', 'A17 (review)',  'A7 (instruction)'. On MT text it is mostly 'A11 (personal)': 0.40 and 'A1 (argumentative)': 0.38.
    - Interview (8): not well connected to just one category: 'A1 (argumentative)': 0.125, 'A11 (personal)': 0.125,   'A12 (promotion)': 0.375,  'A16 (information)': 0.125,  'A17 (review)': 0.25. On MT text, predictions are scattered between 'A1 (argumentative)': 0.25, 'A11 (personal)': 0.375, 'A12 (promotion)': 0.25, and 'A17 (review)': 0.125.
    - List of Summaries/Excerpts (106): most vague category, not connected to FTD categories well - identified as all of the FTD categories; on MT the same.
    - Lyrical (4): half identified as 'A11 (personal)': 0.5, half as  'A4 (fiction)': 0.5; on MT the same combination of labels, but more identified as 'A11 (personal)': 0.75
    - Other (34): as I assumed, the label was predicted with various FTD labels, mostly 'A12 (promotion)': 0.44, 'A16 (information)': 0.15 or  'A17 (review)': 0.21; on MT even more scattered, less promotion ('A12 (promotion)': 0.29)
    - Script/Drama (1): not well predicted, but there is only one instance. Predicted as  'A16 (information)': 1.0; the same on MT

Comparison between the prediction on Slovene and MT text shows that mostly there is not a big difference between prediction on Slovene or English text. On some labels, the predictions are worse on MT (Promotion labels, Prose), on some it is better (Instruction, Research Article, Information/Explanation). On MT, there is a much better identification of News and Opinionated News. In other cases, there is no difference or just a slight difference.

## Compare FTD labels with CORE

In [2]:
# Import the CORE dataset
core_df = pd.read_csv("data-sheets-with-all-info/CORE-all-information.csv", index_col = 0)

core_df.head(3)

label                                               text  split  \
0  NA OP SR OB  The Top TEN 'Whiniest Sets of Fans' in English...  train   
1        NA NE  Ferry consultation needs deeper questions, say...  train   
2        ID DF  I'v been recording and mixing music for about ...  train   

  main_labels sublabels  Len  main_len  sub_len      GINCORE  \
0       NA OP     SR OB    4         2        2  NA OP SR OB   
1         NaN        NE    2         1        1         News   
2          ID        DF    2         1        1        Forum   

         full_names  main_labels_full_names FTD_pred  
0             SR OB                   NA OP      NaN  
1  News Report/Blog               Narrative      NaN  
2  Discussion Forum  Interactive Discussion      NaN

In [3]:
# Filter out only instances that have FTD prediction
core_df = core_df.dropna(subset = ["FTD_pred"])

core_df.shape

(7970, 12)

In [4]:
core_df.columns

Index(['label', 'text', 'split', 'main_labels', 'sublabels', 'Len', 'main_len',
       'sub_len', 'GINCORE', 'full_names', 'main_labels_full_names',
       'FTD_pred'],
      dtype='object')

In [5]:
# Discard uninteresting columns
core_df = core_df[['GINCORE', 'sub_len', 'full_names', 'main_len', 'main_labels_full_names','FTD_pred']]

core_df

GINCORE  sub_len             full_names  main_len  \
6                        Forum        1       Discussion Forum         1   
14                        News        1          Sports Report         1   
16                        News        1       News Report/Blog         1   
23                        News        1       News Report/Blog         1   
25                    NA IN HA        1     Historical Article         2   
...                        ...      ...                    ...       ...   
48419                     News        1       News Report/Blog         1   
48420                 ID HI QA        1  Question/Answer Forum         2   
48423                     News        1       News Report/Blog         1   
48429  Information/Explanation        1   Encyclopedia Article         1   
48446    Opinion/Argumentation        1           Opinion Blog         1   

                      main_labels_full_names            FTD_pred  
6                     Interactive Discussion      A11 (personal)  
14                                 Narrative  A1 (argumentative)  
16                                 Narrative           A8 (news)  
23                                 Narrative  A1 (argumentative)  
25                                     NA IN  A1 (argumentative)  
...                                      ...                 ...  
48419                              Narrative           A8 (news)  
48420                                  ID HI    A7 (instruction)  
48423                              Narrative           A8 (news)  
48429  Informational Description/Explanation   A16 (information)  
48446                                Opinion  A1 (argumentative)  

[7970 rows x 6 columns]

### CORE-main

In [6]:
# Analyse how CORE main labels are connected with FTD labels based on the prediction

# Filter out instances that have multiple main labels
main_labels = core_df[core_df["main_len"] == 1]

pd.crosstab(main_labels["main_labels_full_names"], main_labels["FTD_pred"], normalize="index")

FTD_pred                               A1 (argumentative)  A11 (personal)  \
main_labels_full_names                                                      
How-To/Instructional                             0.043011        0.118280   
Informational Description/Explanation            0.189058        0.020776   
Informational Persuasion                         0.185345        0.021552   
Interactive Discussion                           0.315702        0.289256   
Lyrical                                          0.040650        0.577236   
Narrative                                        0.228284        0.133191   
Opinion                                          0.467799        0.101203   
Spoken                                           0.304348        0.278261   

FTD_pred                               A12 (promotion)  A14 (academic)  \
main_labels_full_names                                                   
How-To/Instructional                          0.093190        0.000000   
Informational Description/Explanation         0.228532        0.052632   
Informational Persuasion                      0.400862        0.000000   
Interactive Discussion                        0.047934        0.009917   
Lyrical                                       0.048780        0.000000   
Narrative                                     0.032002        0.002743   
Opinion                                       0.061571        0.000708   
Spoken                                        0.060870        0.000000   

FTD_pred                               A16 (information)  A17 (review)  \
main_labels_full_names                                                   
How-To/Instructional                            0.003584      0.021505   
Informational Description/Explanation           0.213989      0.089335   
Informational Persuasion                        0.030172      0.258621   
Interactive Discussion                          0.028099      0.041322   
Lyrical                                         0.032520      0.121951   
Narrative                                       0.029259      0.067662   
Opinion                                         0.012739      0.230715   
Spoken                                          0.000000      0.252174   

FTD_pred                               A4 (fiction)  A7 (instruction)  \
main_labels_full_names                                                  
How-To/Instructional                       0.000000          0.713262   
Informational Description/Explanation      0.004848          0.106648   
Informational Persuasion                   0.012931          0.077586   
Interactive Discussion                     0.008264          0.239669   
Lyrical                                    0.089431          0.081301   
Narrative                                  0.017373          0.006400   
Opinion                                    0.012739          0.083510   
Spoken                                     0.017391          0.008696   

FTD_pred                               A8 (news)  A9 (legal)  
main_labels_full_names                                        
How-To/Instructional                    0.000000    0.007168  
Informational Description/Explanation   0.059557    0.034626  
Informational Persuasion                0.004310    0.008621  
Interactive Discussion                  0.014876    0.004959  
Lyrical                                 0.008130    0.000000  
Narrative                               0.480951    0.002133  
Opinion                                 0.029016    0.000000  
Spoken                                  0.078261    0.000000

In [10]:
main_labels["main_labels_full_names"].value_counts()

Narrative                                3281
Informational Description/Explanation    1444
Opinion                                  1413
Interactive Discussion                    605
How-To/Instructional                      279
Informational Persuasion                  232
Lyrical                                   123
Spoken                                    115
Name: main_labels_full_names, dtype: int64

In [11]:
main_labels["main_labels_full_names"].unique()

array(['Interactive Discussion', 'Narrative', 'Opinion',
       'Informational Description/Explanation', 'How-To/Instructional',
       'Lyrical', 'Informational Persuasion', 'Spoken'], dtype=object)

In [12]:
pd.crosstab(main_labels["main_labels_full_names"], main_labels["FTD_pred"], normalize="index").to_dict("index")

{'How-To/Instructional': {'A1 (argumentative)': 0.043010752688172046,
  'A11 (personal)': 0.11827956989247312,
  'A12 (promotion)': 0.0931899641577061,
  'A14 (academic)': 0.0,
  'A16 (information)': 0.0035842293906810036,
  'A17 (review)': 0.021505376344086023,
  'A4 (fiction)': 0.0,
  'A7 (instruction)': 0.7132616487455197,
  'A8 (news)': 0.0,
  'A9 (legal)': 0.007168458781362007},
 'Informational Description/Explanation': {'A1 (argumentative)': 0.18905817174515235,
  'A11 (personal)': 0.02077562326869806,
  'A12 (promotion)': 0.22853185595567868,
  'A14 (academic)': 0.05263157894736842,
  'A16 (information)': 0.21398891966759004,
  'A17 (review)': 0.08933518005540166,
  'A4 (fiction)': 0.004847645429362881,
  'A7 (instruction)': 0.10664819944598337,
  'A8 (news)': 0.05955678670360111,
  'A9 (legal)': 0.03462603878116344},
 'Informational Persuasion': {'A1 (argumentative)': 0.1853448275862069,
  'A11 (personal)': 0.021551724137931036,
  'A12 (promotion)': 0.40086206896551724,
  'A14 

Comparison of main CORE labels and FTD labels:

1. Well-connected:
* 'How-To/Instructional': 'A7 (instruction)': 0.713

2. Not well connected (no clear majority label/majority label does not seem to be appropriate):
* 'Interactive Discussion': mostly 'A1 (argumentative)': 0.315 (percentage of instances of Interactive Discussion, identified as A1) + 'A11 (personal)': 0.289,  'A7 (instruction)': 0.239
* 'Narrative': 'A8 (news)': 0.48, 'A1 (argumentative)': 0.228
* 'Opinion': 'A1 (argumentative)': 0.467, 'A17 (review)': 0.230
* 'Informational Description/Explanation': A12 (promotion)': 0.228, 'A16 (information)': 0.21, 'A1 (argumentative)': 0.189
* 'Lyrical': 'A11 (personal)': 0.577
* 'Informational Persuasion': 'A12 (promotion)': 0.40, 'A1 (argumentative)': 0.185, 'A17 (review)': 0.258
* 'Spoken': 'A1 (argumentative)': 0.30, 'A11 (personal)': 0.278, 'A17 (review)': 0.252




### CORE-sub

In [13]:
core_df.head(2)

GINCORE  sub_len        full_names  main_len  main_labels_full_names  \
6    Forum        1  Discussion Forum         1  Interactive Discussion   
14    News        1     Sports Report         1               Narrative   

              FTD_pred  
6       A11 (personal)  
14  A1 (argumentative)

In [14]:
# Analyse how CORE sub labels are connected with FTD labels based on the prediction

# Filter out instances that have multiple main labels
sub_labels = core_df[core_df["sub_len"] == 1]

pd.crosstab(sub_labels["full_names"], sub_labels["FTD_pred"], normalize="index")

FTD_pred                         A1 (argumentative)  A11 (personal)  \
full_names                                                            
Advertisement                              0.000000        0.000000   
Advice                                     0.172043        0.139785   
Course Materials                           0.173913        0.000000   
Description of a Person                    0.098039        0.058824   
Description of a Thing                     0.139800        0.021398   
Description with Intent to Sell            0.082192        0.018265   
Discussion Forum                           0.374359        0.320513   
Editorial                                  0.923077        0.000000   
Encyclopedia Article                       0.095238        0.009524   
FAQ about How-to                           0.000000        0.000000   
FAQ about Information                      0.060000        0.020000   
Formal Speech                              0.777778        0.111111   
Historical Article                         0.364706        0.000000   
How-to                                     0.057034        0.076046   
Information Blog                           0.290850        0.039216   
Interview                                  0.180851        0.265957   
Legal terms                                0.052632        0.000000   
Letter to Editor                           0.875000        0.125000   
Magazine Article                           0.466667        0.200000   
News Report/Blog                           0.284626        0.016659   
Opinion Blog                               0.660218        0.123337   
Other Forum                                0.000000        0.750000   
Other Information                          0.000000        0.074074   
Other Informational Persuasion             1.000000        0.000000   
Other Opinion                              0.666667        0.000000   
Other Spoken                               0.333333        0.666667   
Personal Blog                              0.122744        0.676895   
Persuasive Article or Essay                0.750000        0.041667   
Poem                                       0.066667        0.200000   
Prayer                                     0.000000        0.000000   
Question/Answer Forum                      0.209524        0.233333   
Reader/Viewer Responses                    0.100000        0.100000   
Recipe                                     0.000000        0.352941   
Religious Blogs/Sermons                    0.671429        0.042857   
Research Article                           0.387879        0.006061   
Reviews                                    0.077778        0.066667   
Short Story                                0.017857        0.160714   
Song Lyrics                                0.037037        0.629630   
Sports Report                              0.163121        0.023050   
TV/Movie Script                            0.000000        0.600000   
Technical Report                           0.375000        0.125000   
Technical Support                          0.000000        0.125000   
Transcript of Video/Audio                  0.833333        0.083333   
Travel Blog                                0.070175        0.438596   

FTD_pred                         A12 (promotion)  A14 (academic)  \
full_names                                                         
Advertisement                           0.666667        0.000000   
Advice                                  0.145161        0.000000   
Course Materials                        0.130435        0.043478   
Description of a Person                 0.143791        0.000000   
Description of a Thing                  0.370899        0.007133   
Description with Intent to Sell         0.465753        0.000000   
Discussion Forum                        0.041026        0.002564   
Editorial                               0.000000        0.000000   
Encyclopedia Article                    0.000000     

In [15]:
sub_labels["full_names"].value_counts()

News Report/Blog                   2101
Opinion Blog                        827
Description of a Thing              701
Sports Report                       564
Personal Blog                       554
Discussion Forum                    390
Reviews                             360
Information Blog                    306
How-to                              263
Description with Intent to Sell     219
Question/Answer Forum               210
Advice                              186
Research Article                    165
Description of a Person             153
Religious Blogs/Sermons             140
Song Lyrics                         108
Encyclopedia Article                105
Interview                            94
Historical Article                   85
Travel Blog                          57
Short Story                          56
FAQ about Information                50
Legal terms                          38
Recipe                               34
Other Information                    27


In [16]:
len(sub_labels.full_names.unique())

44

In [18]:
pd.crosstab(sub_labels["full_names"], sub_labels["FTD_pred"], normalize="index").to_dict("index")

{'Advertisement': {'A1 (argumentative)': 0.0,
  'A11 (personal)': 0.0,
  'A12 (promotion)': 0.6666666666666666,
  'A14 (academic)': 0.0,
  'A16 (information)': 0.0,
  'A17 (review)': 0.0,
  'A4 (fiction)': 0.0,
  'A7 (instruction)': 0.3333333333333333,
  'A8 (news)': 0.0,
  'A9 (legal)': 0.0},
 'Advice': {'A1 (argumentative)': 0.17204301075268819,
  'A11 (personal)': 0.13978494623655913,
  'A12 (promotion)': 0.14516129032258066,
  'A14 (academic)': 0.0,
  'A16 (information)': 0.005376344086021506,
  'A17 (review)': 0.016129032258064516,
  'A4 (fiction)': 0.0,
  'A7 (instruction)': 0.5161290322580645,
  'A8 (news)': 0.005376344086021506,
  'A9 (legal)': 0.0},
 'Course Materials': {'A1 (argumentative)': 0.17391304347826086,
  'A11 (personal)': 0.0,
  'A12 (promotion)': 0.13043478260869565,
  'A14 (academic)': 0.043478260869565216,
  'A16 (information)': 0.30434782608695654,
  'A17 (review)': 0.043478260869565216,
  'A4 (fiction)': 0.0,
  'A7 (instruction)': 0.30434782608695654,
  'A8 (ne

1. Categories that match well:
* 'Advertisement': 'A12 (promotion)': 0.67
* 'Editorial': 'A1 (argumentative)': 0.92
* 'Encyclopedia Article': 'A16 (information)': 0.82
* 'FAQ about How-to':  'A7 (instruction)': 0.66
* 'How-to':  'A7 (instruction)': 0.74
* 'Formal Speech': 'A1 (argumentative)': 0.78
* 'Legal terms': 'A9 (legal)': 0.789
* 'Letter to Editor': 'A1 (argumentative)': 0.875
* 'Opinion Blog': 'A1 (argumentative)': 0.66
* 'Personal Blog: 'A11 (personal)': 0.677
* 'Persuasive Article or Essay': 'A1 (argumentative)': 0.75
* 'Religious Blogs/Sermons': 'A1 (argumentative)': 0.67
* 'Reviews': 'A17 (review)': 0.72
* 'Short Story': 'A4 (fiction)': 0.80
* 'Sports Report': A8 (news)': 0.759
* 'Technical Support': 'A7 (instruction)': 0.875
* Other Informational Persuasion': 'A1 (argumentative)': 1.0
* 'Other Opinion': 'A1 (argumentative)': 0.67
* 'Other Spoken': 'A11 (personal)': 0.67


2. Categories that match, but less well:
* 'Advice': 'A7 (instruction)': 0.51
* 'Description with Intent to Sell': A12 (promotion)': 0.465, 'A17 (review)': 0.283
* 'News Report/Blog': 'A8 (news)': 0.55, 'A1 (argumentative)': 0.28
* 'Recipe': 'A7 (instruction)': 0.47, 'A11 (personal)': 0.35

3. CORE sub categories with no (appropriate) majority FTD label:
* 'Course Materials': 'A7 (instruction)': 0.30 or 'A16 (information)': 0.30
* 'Description of a Person':  'A16 (information)': 0.33, 'A17 (review)': 0.228
* 'Description of a Thing': 'A12 (promotion)': 0.37
* 'Discussion Forum': 'A1 (argumentative)': 0.37, 'A11 (personal)': 0.32
* 'FAQ about Information': 'A7 (instruction)': 0.44, 'A12 (promotion)': 0.28
* 'Historical Article': 'A16 (information)': 0.423, 'A1 (argumentative)': 0.36
* 'Information Blog': 'A1 (argumentative)': 0.29, 'A7 (instruction)': 0.21
* 'Interview': ''A17 (review)': 0.36, A1 (argumentative)': 0.18, 'A11 (personal)': 0.265
* 'Magazine Article': 'A1 (argumentative)': 0.47
* 'Poem': 'A4 (fiction)': 0.33, A11 (personal)': 0.2, 'A17 (review)': 0.27
* 'Prayer': A16 (information)': 0.67
* 'Question/Answer Forum':  'A7 (instruction)': 0.357, 'A11 (personal)': 0.23, A1 (argumentative)': 0.209
* 'Reader/Viewer Responses': 'A17 (review)': 0.3, 'A12 (promotion)': 0.3, 'A7 (instruction)': 0.2
* 'Research Article': 'A1 (argumentative)': 0.39, 'A14 (academic)': 0.376
* 'Song Lyrics': 'A11 (personal)': 0.63 (matches well, but debatable if it's appropriate category)
* 'TV/Movie Script': A11 (personal)': 0.6
* 'Technical Report': 'A1 (argumentative)': 0.375, 'A7 (instruction)': 0.25 
* 'Transcript of Video/Audio': 'A1 (argumentative)': 0.83
* 'Travel Blog': 'A11 (personal)': 0.438, A17 (review)': 0.25, 'A12 (promotion)': 0.19
* 'Other Information':  'A7 (instruction)': 0.30, 'A16 (information)': 0.185